contracts/DAORegistryScheme.sol

```diff
@@ -0,0 +1,181 @@
+pragma solidity >=0.5.4 <0.6.0;
+
+import "@daostack/arc/contracts/universalSchemes/UniversalScheme.sol";
+import "@daostack/arc/contracts/votingMachines/VotingMachineCallbacks.sol";
+import "@daostack/infra/contracts/votingMachines/IntVoteInterface.sol";
+import "@daostack/infra/contracts/votingMachines/VotingMachineCallbacksInterface.sol";
+import "./DAORegistry.sol";
+
+/**
+ * @title A universal scheme for letting organizations manage a registrar of named DAOs
+ * @dev The DAORegistryScheme has a registry of DAOs for each organization.
+ *      The organizations can through a vote choose to register allowing them to add/remove names inside the registry.
+ */
+contract DAORegistryScheme is UniversalScheme, VotingMachineCallbacks, ProposalExecuteInterface {
+
+    event NewDAOProposal (
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address _proposedAvatarAddress,
+        string _proposedAvatarName
+    );
+
+    event RemoveDAOProposal (
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address _proposedAvatarAddress
+    );
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _param, bytes _returnValue);
+    event ProposalDeleted(address indexed _avatar, bytes32 indexed _proposalId);
+
+    // a DAORegistryProposal is a proposal to add or remove a named DAO to/from an organization
+    struct DAORegistryProposal {
```

Comment:
So i think this is an elegant pattern, and it could be abstracted in a separate "base" generic action class. But having this structurewithin a contract that just allows two different function calls ("add" and "remove"), I think it introduces unecessary complexity: why not just call the functions directly instead of passin through this abstraction?